EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs

```diff
@@ -0,0 +1,207 @@
+﻿/**
+ * Copyright 2018-2021 MobiledgeX, Inc. All rights and licenses reserved.
+ * MobiledgeX, Inc. 156 2nd Street #408, San Francisco, CA 94105
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using UnityEngine;
+
+namespace EdgeMultiplay
+{
+    /// <summary>
+    /// Use for syncing GameObjects local position and/or local rotation
+    /// </summary>
+    public class EdgeMultiplayObserverLocal : EdgeMultiplayObserver
+    {
+        #region EdgeMultiplayObserverLocal Variables
+
+        private Vector3 latestPosition;
+        private Vector3 latestRotation;
+        private bool LocalPlayeIsMaster;
+        private NetworkedPlayer networkPlayer;
+        private Vector3 targetPosInterpolated;
+        private Vector3 targetRotInterpolated;
+
+        #endregion
+
+        #region MonoBehaviour Callbacks
+
+#if UNITY_EDITOR
+
+        //Reset is a Monobehaviour callback and is called once a Component is added to a GameObject
+        [ExecuteAlways]
+        void Reset()
+        {
+            eventId = Guid.NewGuid().ToString("N") + gameObject.GetInstanceID().ToString();
+        }
+
+#endif
+        private void Start()
+        {
+            if (!attachedToPlayer)
+            {
+                LocalPlayeIsMaster = EdgeManager.localPlayerIsMaster;
+            }
+            EdgeManager.observers.Add(this);
+            networkPlayer = GetComponent<NetworkedPlayer>();
+            positionFromServer = transform.localPosition;
+            rotationFromServer = transform.localRotation.eulerAngles;
+        }
+
+        private void Update() 
+        {
+            if (EdgeManager.gameStarted)
+            {
+                if ((latestPosition != transform.localPosition && syncOption == SyncOptions.SyncPosition)
+                || (latestRotation != transform.localRotation.eulerAngles && syncOption == SyncOptions.SyncRotation)
+                || ((latestRotation != transform.localRotation.eulerAngles || latestPosition != transform.localPosition) && syncOption == SyncOptions.SyncPositionAndRotation))
```

Comment:
These conditionals are a little confusing. Might be useful to have a helper function: bool RequiresUpdate() that either has a switch statement or a series of if elses.

eg.
```
bool RequireUpdate()
{ 
  switch syncOption
  {
    case SyncOptions.SyncPosition:
      if (latestPosition != transform.locationPosition) return true;
    case SyncOptions.SyncRotation:
    case SyncOptions.SyncPositionAndRotation:
  }
}
```

Reply:
++

Reply:
Done, Many thanks for the review